shuup/admin/modules/manufacturers/object_selector.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2021, Shuup Commerce Inc. All rights reserved.
+#
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+from typing import Iterable, Tuple
+
+from shuup.admin.utils.permissions import has_permission
+from shuup.admin.views.select import BaseAdminObjectSelector
+from shuup.core.models import Manufacturer
+
+
+class ManufacturerAdminObjectSelector(BaseAdminObjectSelector):
+    ordering = 6
+
+    @classmethod
+    def handles_selector(cls, selector):
+        return selector == "shuup.manufacturer"
+
+    def has_permission(self, user):
+        return has_permission(user, "manufacturer.object_selector")
+
+    def get_objects(self, search_term, *args, **kwargs) -> Iterable[Tuple[int, str]]:
+        """
+        Returns an iterable of tuples of (id, text)
+        """
+        objects = list(
```

Comment:
Any business logic to add here?

Reply:
manufacturers are shared by all shops, it means only staff users should access them. let the permissions handle it.